src/Contacts/src/ContactListParser.cpp

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file ContactListParser.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <fstream>
+#include <string_view>
+
+#include <nlohmann/json.hpp>
+
+#include <manif/Rn.h>
+#include <manif/SE3.h>
+#include <manif/SO3.h>
+
+// BipedalLocomotion
+#include <BipedalLocomotion/Contacts/ContactList.h>
+#include <BipedalLocomotion/Contacts/ContactListParser.h>
+#include <BipedalLocomotion/TextLogging/Logger.h>
+
+namespace BipedalLocomotion
+{
+namespace Contacts
+{
+
+ContactListMap contactListMapFromJson(const std::string& filename)
+{
+    constexpr auto logPrefix = "[contactListMapFromJson]";
+    ContactListMap map;
+
+    // open the file
+    std::ifstream inputStream(filename);
+
+    if (!inputStream.is_open())
+    {
+        log()->error("{} Failed to open {}. An empty map will be returned.", logPrefix, filename);
+        return map;
+    }
+
+    nlohmann::json jsonFile;
+    inputStream >> jsonFile;
+
+    // iterate over the name of the contacts
+    PlannedContact tempContact;
+    for (auto& [key, val] : jsonFile.items())
+    {
+        try
+        {
+            const std::vector<double>& contactPosition = val.at("position");
+            const std::vector<double>& contactQuaternion = val.at("quaternion");
+
+            if (contactPosition.size() != 3 || contactQuaternion.size() != 4)
+            {
+                log()->error("{} The size of the position and/or the quaternion is different from "
+                             "expected. Position size {}, expected: 4. Quaternion size: {}, "
+                             "expected: 4. An empty ContactListMap is returned",
+                             logPrefix,
+                             contactPosition.size(),
+                             contactQuaternion.size());
+                return ContactListMap();
+            }
+
+            tempContact.pose.quat(Eigen::Map<const manif::SO3d>(contactQuaternion.data()));
+            tempContact.pose.translation(Eigen::Map<const manif::R3d>(contactPosition.data()).coeffs());
+            tempContact.activationTime = val.at("activation_time");
+            tempContact.deactivationTime = val.at("deactivation_time");
```

Comment:
Maybe it is worth adding somewhere the expected format

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/316/commits/b7be6c963ca04999a14e4a46aaf752862463ed9e